orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/RunTaskHandler.kt

```diff
@@ -97,35 +98,54 @@ open class RunTaskHandler
         val exceptionDetails = exceptionHandlers.shouldRetry(e, taskModel.name)
         if (exceptionDetails?.shouldRetry ?: false) {
           log.warn("Error running ${message.taskType.simpleName} for ${message.executionType.simpleName}[${message.executionId}]")
-          queue.push(message, task.backoffPeriod())
-          trackResult(stage, task.javaClass, RUNNING)
+          queue.push(message, task.backoffPeriod(taskModel))
+          trackResult(stage, taskModel, RUNNING)
         } else if (e is TimeoutException && stage.getContext()["markSuccessfulOnTimeout"] == true) {
           queue.push(CompleteTask(message, SUCCEEDED))
         } else {
           log.error("Error running ${message.taskType.simpleName} for ${message.executionType.simpleName}[${message.executionId}]", e)
           stage.getContext()["exception"] = exceptionDetails
           repository.storeStage(stage)
           queue.push(CompleteTask(message, stage.failureStatus()))
-          trackResult(stage, task.javaClass, stage.failureStatus())
+          trackResult(stage, taskModel, stage.failureStatus())
         }
       }
     }
   }
 
-  private fun trackResult(stage: Stage<*>, taskType: Class<Task>, status: ExecutionStatus) {
+  private fun trackResult(stage: Stage<*>, taskModel: com.netflix.spinnaker.orca.pipeline.model.Task, status: ExecutionStatus) {
     val id = registry.createId("task.invocations")
       .withTag("status", status.toString())
       .withTag("executionType", stage.getExecution().javaClass.simpleName)
-      .withTag("stageType", stage.getType())
-      .withTag("taskType", taskType.simpleName)
```

Comment:
Not hugely ... and it cuts down on the amount of time we can analyze metrics.

Once we know the hot application we can just look at it's tasks/pipelines for details?